pyogrio/tests/conftest.py

```diff
@@ -27,12 +27,16 @@ def prepare_testfile(testfile_path, dst_dir, ext):
     dst_path = dst_dir / f"{testfile_path.stem}{ext}"
     if dst_path.exists():
         return dst_path
-    gdf = pyogrio.read_dataframe(testfile_path)
+
+    meta, _, geometry, field_data = read(testfile_path)
+
     if ext == ".fgb":
         # For .fgb, spatial_index=False to avoid the rows being reordered
-        pyogrio.write_dataframe(gdf, dst_path, spatial_index=False)
-    else:
-        pyogrio.write_dataframe(gdf, dst_path)
+        meta["spatial_index"] = False
+        # allow mixed Polygons/MultiPolygons type
+        meta["geometry_type"] = "Unknown"
```

Comment:
This one line is a change compared to what was there before? Or did `write_dataframe` infer such a geometry type? (I would have expected that it would infer "MultiPolgyon" and set promote_to_multi to true)

Reply:
The detection and promotion happens in `write_geodataframe` and doesn't happen in the case of the raw `write`.

With a little bit of refactoring we could move that into `write` instead (I think this was even suggested when that was first added), but I didn't think we needed to do so prior to 0.4 (changes noted in the changelog are specific to GeoDataFrame)